database.hh

```diff
@@ -870,7 +879,19 @@ public:
     }
 
     bool compaction_enforce_min_threshold() const {
-        return _config.compaction_enforce_min_threshold;
+        return _config.compaction_enforce_min_threshold || _is_bootstrap_or_replace;
+    }
+
+    unsigned min_compaction_threshold() {
+        // During receiving stream operations, the less we compact the faster streaming is. For
+        // bootstrap and replace thereThere are no readers so it is fine to be less aggressive with
+        // compactions as long as we don't ignore them completely (this could create a problem for
+        // when streaming ends)
+        if (_is_bootstrap_or_replace) {
+            return std::max(schema()->min_compaction_threshold(), 16);
```

Comment:
I think @raphaelsc is right that we should consider max_threshold here. A reason to set it is if we have very large rows and therefore can run out of memory; consider 32MiB rows, with 16 sstables we need 512MiB per compaction.

Reply:
I still don't see what the problem is.
His concern was that "something wrong" (unspecified) would happen if max_threshold was lower than min_threshold. I tested that, and in that case max_threshold is still respected: we compact a `max_threshold` set.

So if max_threshold is higher than min_threshold, it does what everyone expects and caps the amount of SSTables in the set.

If max_threshold is lower than min_threshold, it is still respected since that is how many SSTables we will compact. Whichever way, max_threshold is respected.

So what do you have in mind here? Are you asking to make 16 even lower when max_threshold is lower?

Reply:
Yes, is (say) max_threshold was set to 4 to avoid problems with 32Gib rows.

Reply:
Ok, I will use max_threshold / 2 as a goal instead of 16.

Reply:
I force-pushed a version with this

Reply:
And if max_threshold == 2, you will not make progress.

I suggest min(16, max_threshold)

Reply:
Ok, will change.

setting the target to `min(16, max_threshold)`, and then using that target as `min_threshold 
= max(target, min_threshold)`

This is only ill-defined if `max_threshold < min_threshold`. However that is not a valid configuration (although we don't enforce it), and as I tested, all that means is that max_threshold becomes both min and max thresholds.

An interesting exercise to the reader is to set `min_threshold` to 1 and see how fun that is.